[UnifiedPDF] PDFPluginBase should send cursor updates to the web page

aprotyas opened the pull request:
#### ce5a85abac3927874650474cd432d10cbd3ca083
<pre>
[UnifiedPDF] PDFPluginBase should send cursor updates to the web page
<a href="https://bugs.webkit.org/show_bug.cgi?id=264913">https://bugs.webkit.org/show_bug.cgi?id=264913</a>
<a href="https://rdar.apple.com/118483364">rdar://118483364</a>

Reviewed by Tim Horton.

In preparation to get cursor updates working for `UnifiedPDFPlugin`, we
should move `PDFPlugin::notifyCursorChanged` up into the `PDFPluginBase`
structure. We should be doing the same (trivial) amount of work for both
PDF plugin types to send cursor updates to the web page.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(toWebCoreCursorType):
(-[WKPDFLayerControllerDelegate setMouseCursor:]):
(WebKit::PDFPlugin::notifySelectionChanged):
(WebKit::coreCursor): Deleted.
(WebKit::PDFPlugin::notifyCursorChanged): Deleted.
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
(WebKit::PDFPluginBase::notifyCursorChanged):

Canonical link: <a href="https://commits.webkit.org/270863@main">https://commits.webkit.org/270863@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3cd8be781cf3be3499a9aaf7988b1bfe3f94c6d8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26596 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5211 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27843 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28808 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24323 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/7092 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2618 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24282 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26857 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/4093 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22862 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/3557 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/3637 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29293 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24307 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24251 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29869 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3635 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1837 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27766 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/5073 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6399 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4109 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3966 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
```diff
@@ -28,14 +28,17 @@
 #import <WebCore/AXObjectCache.h>
 #import <WebCore/ArchiveResource.h>
 #import <WebCore/Chrome.h>
+#import <WebCore/Cursor.h>
```
This import is redundant now, but if you make the suggested change in https://github.com/WebKit/WebKit/pull/20561, it should be kept.

aprotyas replied:
Yeah that's a good point. I'll make it `WebCore::CursorType` and not pull in the `WebCore/Chrome.h` header in `PDFPluginBase.h`.

aprotyas committed:
[UnifiedPDF] PDFPluginBase should send cursor updates to the web page
https://bugs.webkit.org/show_bug.cgi?id=264913
rdar://118483364

Reviewed by Tim Horton.

In preparation to get cursor updates working for `UnifiedPDFPlugin`, we
should move `PDFPlugin::notifyCursorChanged` up into the `PDFPluginBase`
structure. We should be doing the same (trivial) amount of work for both
PDF plugin types to send cursor updates to the web page.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(toWebCoreCursorType):
(-[WKPDFLayerControllerDelegate setMouseCursor:]):
(WebKit::PDFPlugin::notifySelectionChanged):
(WebKit::coreCursor): Deleted.
(WebKit::PDFPlugin::notifyCursorChanged): Deleted.
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
(WebKit::PDFPluginBase::notifyCursorChanged):

Canonical link: https://commits.webkit.org/270863@main
